static/js/cf-frontend.js

```diff
@@ -346,15 +348,33 @@ function selectedCheckerOnChange() {
 
 
 }
+
+function linkGen(){
+  var checker_value = $("#type_system").val();
+  var typeSystemURL = encodeURI(checker_value);
+  typeSystemURL = "#typeSystem=" + typeSystemURL;
+
+  var input = pyInputGetValue(); 
+  var inputURL = encodeURI(input);
+  inputURL = "&code=" + inputURL;
+
+  var curUrl = window.location.host + '/';
+  document.getElementById("link").value = (curUrl + typeSystemURL + inputURL);
```

Comment:
Also, `link` doesn't self-explained enough, please change to some name more clearer.